### ISSUE TITLE
auth_id format

### ISSUE THREAD

**Issue by piotrpl** [ocpi/ocpi#131]
`auth_id` which is part of the `Token` object is being used to uniquely identify the EV Driver contract within the eMSPs platform. 
Given the above description, this id can be any value that is unique within the eMSP's platform and depending on the eMSP's system it can be of various format. Having the eMI3 recommendation in the definition of this field is a bit confusing and misleading I believe.
Difficulty of using eMI3 format was especially visible during integration with e-clearing.net. Not all of the eMPSs are able to provide contract id in the compliant format.

**Comment by piotrpl** [comment]
Another question which falls into this very same bucket is the reason for using `auth_id` at all. Token object comes with the `uid` which can be used to uniquely identify it across CPOs and eMSPs. It could be used to bind `CDRs` and `Sessions` together with the `Token`.
The main difference is that `auth_id` allows to identify the same driver across different `Tokens` but this should be taken care of by the eMSPs system not outside of it.
Exposing contract id via `auth_id` may be considered also as unwanted/security concern.

**Comment by RobertdeLeeuw-IHomer** [comment]
@piotrpl This is the OCPI specification. Parties can always make different business agreements. I guess that e-Clearing requires tokens to be eMI3. But that would then be an additional requirement on top of OCPI. Which can be fine if all parties agree with that. 

Some party(s) pointed out that they needed to support a use case where a Driver switches MSP but keeps his own RFID token. So the `uid` will then move from one system to another. 

And the uid is only unique if it is an RFID token. We might start using other ways of authorization...

**Comment by piotrpl** [comment]
@RobertdeLeeuw-IHomer yes I understand this is OCPI specification. The reason why I brought up eMI3 is that it is mentioned as a recommendation for thise `auth_id` field, but the field definition is not aligned with the recommendation which IMO makes it unclear.
I think that in it would make more sense to have `uid` field being used like in other objects where it serves as unique identifier for an object. In the case of `Token` this could be `contract` or `UUID` or any other value that makes sense to eMSP. `auth_id` then could provide the "token" be it RFID hidden ID or any other authentication value.

**Comment by RobertdeLeeuw-IHomer** [comment]
The Token object is causing a lot confusion.
Maybe adding a real ID field might solve this.

Some proposed to change the uid field to hidden_id. That is fine for RFID tokens, but the token object can also be used for non RFID tokens, then this is not the hidden_id, but the tokens unique id.

**Comment by piotrpl** [comment]
I think that the confusion may come from the fact that it is different than others. I would really opt for using `uid` as eMSP's internal id, just like in other objects and `auth_id` as a special value for authentication. 
It actually makes sense as whether it is RFID hidden_id, credit card number or anything else you use it for authorization.

**Comment by RobertdeLeeuw-IHomer** [comment]
We discussed this during the Strasbourg meeting: Do we need the auth_id/contract_id/eMI3 ID at all in OCPI? Most don't see a reason to have it. But organisations like e-Violin require it.
So for 2.2 we decided to keep it. It is on the list for 3.0 to see if we can remove it.
